### PR TITLE
handle error case better when facts return no results back

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: src/nixos-remote.sh [hunk 1]
```diff
@@ -50,6 +50,8 @@ nix_options=(
 
 declare -A disk_encryption_keys
 
+set -eu -o pipefail
```
Comment (Kranzes) [inline src/nixos-remote.sh:53]: This is added to the wrapped script with writeShellApplication, it probably doesn't harm to have it twice but what is the reason for doing it?

Comment (Mic92) [reply src/nixos-remote.sh:53]: I forgot that this is already done there.

**Review by phaer — COMMENTED** [review]
Aborting when fact collection fails is surely a good idea.

But whats up with removing the colons? If i am not mistaken, the only difference the colon would make would be to handle i.e. empty strings the same as undefined variable. Isn't that the right thing to do, at least for flakeAttrs & so?

**Comment by Mic92** [comment]
> Aborting when fact collection fails is surely a good idea.
> 
> But whats up with removing the colons? If i am not mistaken, the only difference the colon would make would be to handle i.e. empty strings the same as undefined variable. Isn't that the right thing to do, at least for flakeAttrs & so?

shfmt did this when adding `set -eux -o pipefail`, so I assume it's fine.

**Comment by phaer** [comment]
Oh okay, thought that was a deliberate choice. But shouldn't matter too much in this case, because one would hopefully not set them to a null value

**Comment by phaer** [comment]
bors merge

**Comment by Mic92** [comment]
bors merge

**Comment by Mic92** [comment]
bors merge

**Comment by Mic92** [comment]
bors merge

**Comment by Mic92** [comment]
bors merge

**Pull request merged**